libs/api/libbpf_program.cpp

```diff
@@ -139,6 +140,45 @@ bpf_load_program(
     return bpf_load_program_xattr(&load_attr, log_buf, log_buf_sz);
 }
 
+int
+bpf_prog_load(
+    enum bpf_prog_type prog_type,
+    const char* prog_name,
+    const char* license,
+    const struct bpf_insn* insns,
+    size_t insn_cnt,
+    const struct bpf_prog_load_opts* opts)
+{
+    UNREFERENCED_PARAMETER(license);
+
+    const ebpf_program_type_t* program_type = _get_ebpf_program_type(prog_type);
+    if (program_type == nullptr) {
+        return libbpf_err(-EINVAL);
+    }
+
+    if ((insn_cnt == 0) || (insn_cnt > UINT32_MAX / sizeof(ebpf_inst))) {
```

Comment:
I think maximum instruction count limit is UINT_MAX. Libbpf also has the same limit:
`if (insn_cnt > UINT_MAX)`

But here we have the limit set to `UINT32_MAX / sizeof(ebpf_inst)`.

Reply:
This is due to an existing limitation of ebpf_program_load_bytes(). not new in this PR.   As shown in line 172, ebpf_program_load_bytes() requires the byte count size to fit in uint32_t and so requires a smaller maximum.   Feel free to file an issue to fix ebpf_program_load_bytes() to support UINT_MAX * sizeof(ebpf_inst).

Reply:
Filed #1145